### ISSUE TITLE
Confused about normal map rendering

### ISSUE THREAD

**Issue by edoRemelli** [B1ueber2y/DIST-Renderer#2]
Thanks for sharing your work with the community!

I have a quick question: in the original paper, in Eq. 3, you mention that normal maps are computed using finite differences on the deep signed distance function. 

However, when digging through your code, it seems like normals are rather either computed analytically from the deep signed distance field (use_depth2normal = False) or are computed using finite differences on the depth map (use_depth2normal = True). 

I was curious to know what is your insight on these 3 alternative approaches to compute the same quantity!

**Comment by B1ueber2y** [comment]
Yes we set the use_depth2normal=False in our default setting, which means that we use the finite differences of the deep signed distance function. You cannot compute the gradients of the surface normal analytically in a trivial manner since it needs you to make some changes to the deep learning framework. We use the PyTorch autograd feature, which indeed uses the finite differences to compute gradients.

For the use_depth2normal=True, it is just an alternative to directly compute the surface normal from the depth. Actually it is just another possible solution that is not mentioned in the paper and not used in the default setting. We tried it in the developing process and it seems that the estimated normal is not that accurate compared to that computed from the finite differences of the SDF. So we set it to false in the default setting but just keeps this intermediate solution as an option in the renderer.

Thanks!
